### PR TITLE
add ABS and SIGN functions

### PR DESCRIPTION
**Pull request by earney**
I noticed there is another pull request for adding ABS and SIGN, and thought I would try an implementation myself.
This also resolves issue #283

### CODE REVIEW

**Review by ever0de — APPROVED** [review]
LGTM
Let's wait for the maintainer's review.

**Review by ever0de — COMMENTED** [review]
Can you add test cases related to ABS and SIGN to the `test-suite` crates?
Check this: <https://github.com/gluesql/gluesql/tree/main/test-suite/src/function>

File: core/src/ast/function.rs
Comment (panarch) [inline core/src/ast/function.rs:10]: Let's simply follow the Rust's naming patterns.
`Abs` would be good to use for this.

File: test-suite/src/function/abs.rs
Comment (panarch) [inline test-suite/src/function/abs.rs:19]: sql is case-insensitive, so there's no problem to execute this statement.
However we used uppercase letters for all other keyword `SELECT`, `AS` and `FROM` only except for `abs` which is also the same keyword like others.
It looks better to use uppercase letters for all keyword for the consistency.

**Review by panarch — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution, welcome 👍 👍 👍  

There are.. a few small requests mainly about tests.
0. Two inline comments
1. Let's also have integration tests for `SIGN` function.
2. To make newly added integration tests to work, you also need to edit these two files
[test-suite/src/function/mod.rs](https://github.com/gluesql/gluesql/blob/main/test-suite/src/function/mod.rs)
[test-suite/src/lib.rs](https://github.com/gluesql/gluesql/blob/main/test-suite/src/lib.rs)
Then, `cargo test abs` or `cargo test sign` will show that new tests work.

File: core/src/ast/function.rs
Comment (earney) [reply core/src/ast/function.rs]: Thanks..  will make necessary modifications.

**Comment by earney** [comment]
I have pushed the changes.  Please let me know if I need to make additional modifications.
I did find one issue.
when someone executes a query such as:  select -0 from mytable;  The -0 automatically gets modified to 0.  I'm not sure if this is part of the sql standard but this does affect a few of my test cases, which I had to give the responses it was looking for instead of the correct response.

**Comment by earney** [comment]
> 

These suggestions make sense to me.  I will look into implementing them.  Thanks for the feedback!

**Comment by earney** [comment]
This pull request should be ready to merge with main.  If you find other issues (or if I missed an issue), please let me know.

**Review by ever0de — APPROVED** [review]
Looks good thanks!

File: core/src/executor/evaluate/function.rs
Comment (panarch) [inline core/src/executor/evaluate/function.rs:224]: ❤️

**Review by panarch — APPROVED** [review]
Looks all great!! Thanks a lot 👍 👍 👍

Until the first PR merged, to run GitHub Actions, the maintainer needs to approve to run so it takes some time to check whether your commit pass the tests or not.
From now on, your first PR has merged :) so, Action will be triggered just after you push the new code to PRs.

**Pull request merged**
